src/encoded/static/components/genetic_modification.js

```diff
@@ -129,15 +145,18 @@ ModificationSiteItems.defaultProps = {
 // render into the GM summary panel as its own section;
 const ModificationSite = (props) => {
     const { geneticModification } = props;
-    const itemClass = globals.itemClass(geneticModification, 'view-detail key-value');
 
-    return (
-        <div>
-            <hr />
-            <h4>Modification site</h4>
-            <ModificationSiteItems geneticModification={geneticModification} itemClass={itemClass} />
-        </div>
-    );
+    if (hasModificationSiteProps(geneticModification)) {
+        const itemClass = globals.itemClass(geneticModification, 'view-detail key-value');
+        return (
+            <div>
+                <hr />
+                <h4>Modification site</h4>
+                <ModificationSiteItems geneticModification={geneticModification} itemClass={itemClass} />
+            </div>
+        );
+    }
+    return null;
```

Comment:
Not big deals
1. You may need to comment on why you are returning null. Not everyone may realize you need null with components instead of undefined (which functions/methods return by default).

2. My understanding is that code should check if it can do what it needs to do early. If it detects it cannot, then it does not try to proceed. An alternate approach would be to check for (!hasModificationSiteProps(geneticModification)). If this true, return null. If not true, the if statement will be skipped and the code will proceed as normal.

Reply:
For #2, what would the difference be? In your version, has hasModificationSiteProps would be called and if it returns false, the rest of the code is skipped and null is returned. In my version, isn’t the same thing true? None of the code would be executed in my version either.

Reply:
For #1, I think people understand that, and we do that _everywhere_. React also says to do this, so people can just refer to the [React documentation](https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering). We would have to add this same comment to about 90 places in our code.